packages/web-components/src/components/fade-in-out/fade-in-out.ts

```diff
@@ -0,0 +1,253 @@
+        forEach(this.querySelectorAll(selectorTargets), item => {
+          this._rootObserver?.observe(item);
+        });
+      }
+      this._resizeObserver.observe(this.ownerDocument!.documentElement);
+    }
+  }
+
+  /**
+   * Cleans-up and creats the root intersection observer.
+   *
+   * @param [options] The options.
+   * @param [options.create] `true` to create the new intersection observer.
+   */
+  private _cleanAndCreateRootObserver({ create }: { create?: boolean } = {}) {
+    if (this._rootObserver) {
+      this._rootObserver.disconnect();
+      this._rootObserver = null;
+    }
+
+    if (create) {
+      this._rootObserver = new IntersectionObserver(this._handleExit);
+    }
+  }
+
+  /**
+   * Cleans-up and creats the inner intersection observer.
+   *
+   * @param [options] The options.
+   * @param [options.create] `true` to create the new intersection observer.
+   * @param [options.viewportMargin] recalculated margin value for the observer
+   */
+  private _cleanAndCreateInnerObserver({ create, viewportMargin }: { create?: boolean; viewportMargin?: string } = {}) {
+    if (this._innerObserver) {
+      this._innerObserver.disconnect();
+      this._innerObserver = null;
+    }
+
+    if (create) {
+      this._innerObserver = new IntersectionObserver(this._handleEntrance.bind(this), { rootMargin: viewportMargin });
+      const { selectorTargets } = this;
+      if (selectorTargets) {
+        forEach(this.querySelectorAll(selectorTargets), item => {
+          this._innerObserver?.observe(item);
+        });
+      }
+    }
+  }
+
+  /**
+   * Cleans-up and creats the resize observer.
+   *
+   * @param [options] The options.
+   * @param [options.create] `true` to create the new resize observer.
+   */
+  private _cleanAndCreateResizeObserver({ create }: { create?: boolean } = {}) {
+    if (this._resizeObserver) {
+      this._resizeObserver.disconnect();
+      this._resizeObserver = null;
+    }
+
+    if (create) {
+      // TODO: Wait for `.d.ts` update to support `ResizeObserver`
+      // @ts-ignore
+      this._resizeObserver = new ResizeObserver(this._handleResize.bind(this));
+    }
+  }
+
+  /**
+   * Handler to add recalculated rootMargin to observer upon resize.
+   *
+   * This calculation is done to retrieve the best fitting top and bottom
+   * margin for the fade animation to trigger/remove from elements in a
+   * user's screen.
+   *
+   * The resulting value is the optimal point where a user's attention will be
+   * grabbed by the animation without restricting their view and perception of
+   * the adopting website. The displayed elements will keep the user's attention
+   * for a longer time as they scroll down the website.
+   *
+   * @private
+   */
+  private _handleResize() {
+    this._cleanAndCreateInnerObserver({
+      create: true,
+      viewportMargin: `-${((document.documentElement.clientHeight * _colSpan) / breakpoints.max.columns).toString()}px 0px`,
```

Comment:
```suggestion
      viewportMargin: `-${((this.ownerDocument!.documentElement.clientHeight * _colSpan) / breakpoints.max.columns).toString()}px 0px`,
```

Reply:
ah this totally slipped through me, thanks for noticing!